kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/ConsumerRecord.java

```diff
@@ -51,6 +52,28 @@ public Map<String, String> getRecordHeaders() {
         return recordHeaders;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ConsumerRecord that = (ConsumerRecord) o;
+        return Objects.equals(topic, that.topic) &&
+                Objects.equals(key, that.key) &&
+                Objects.equals(value, that.value) &&
+                partition == that.partition &&
+                offset == that.offset &&
+                recordHeaders.equals(that.recordHeaders);
```

Comment:
you want to consider the element equality.
```suggestion
                Objects#deepEquals(recordHeaders, that.recordHeaders);
```